### PR TITLE
Feature Update : Images getting deleted from S3 bucket on deletion of…

### PR DESCRIPTION
**Pull request by amit-crownstack**
… Lessions,Cource

## PR Checklist

- [x] I have reviewed the pull request myself
- [x] I have tested the feature myself manually
- [ ] I have updated JIRA
- [x] I have added the description to PR which tells the reviewer what this PR is all about
- [x] I have made good commit messages
- [x] I am not mixing two different types of work in one PR. (Do not mix two different features into one PR, or refactoring and feature into one PR)

### CODE REVIEW

File: apps/quick-learn-backend/src/routes/course/course.service.ts [hunk 1]
```diff
@@ -320,6 +354,12 @@ export class CourseService extends BasicCrudService<CourseEntity> {
       throw new BadRequestException(en.CourseNotFound);
     }
 
+    const imageUsed = await this.getLessonsInCource({ id } , ['lessons']); // Get course without lessons to delete all images from s3
+
+    // console.log(course.imageUsed , 'course');
```
Comment (Snehdeep-ts) [inline apps/quick-learn-backend/src/routes/course/course.service.ts:359]: remove this

File: apps/quick-learn-backend/src/file/fileService.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Injectable } from "@nestjs/common";
+import { S3Client, DeleteObjectCommand } from '@aws-sdk/client-s3';
+
+@Injectable()
+export class FileService {
+    private s3Client: S3Client;
+    static deleteFiles: unknown;
+
+    constructor() {
+        this.s3Client = new S3Client({
```
Comment (shivraj-bhagat) [inline apps/quick-learn-backend/src/file/fileService.service.ts:10]: update this to use config services

File: apps/quick-learn-backend/src/routes/course/course.service.ts [hunk 3]
```diff
@@ -155,6 +158,37 @@ export class CourseService extends BasicCrudService<CourseEntity> {
     return course;
   }
 
+   /**
+   * Gets lessions details within cource with relations
+   */
+
+   async getLessonsInCource(
```
Comment (shivraj-bhagat) [inline apps/quick-learn-backend/src/routes/course/course.service.ts:165]: function name should depict the function output/context in a better way.

File: apps/quick-learn-backend/src/common/utils/helper.ts [hunk 4]
```diff
@@ -47,4 +47,44 @@ export default class Helpers {
       ? sanitizedContent.substring(0, 250) + '...'
       : sanitizedContent;
   }
+
```
Comment (shivraj-bhagat) [inline apps/quick-learn-backend/src/common/utils/helper.ts:50]: add a jsdoc comment

File: apps/quick-learn-backend/src/common/utils/helper.ts [hunk 5]
```diff
@@ -47,4 +47,44 @@ export default class Helpers {
...
+      let match: RegExpExecArray | null;
+  
+      // Execute regex to find all matches
+      while ((match = regex.exec(input as string)) !== null) {
+        // Add the URL found to the array
+        if (match[1]) {
+          urls.push(match[1]);
+        }
+      }
+    } else {
```
Comment (shivraj-bhagat) [inline apps/quick-learn-backend/src/common/utils/helper.ts:71]: If we're making this label an optional parameter, then we should handle this over here as well. There should not be data called item[undefined].

**Review by shivraj-bhagat — APPROVED** [review]
lgtm

**Pull request merged**